var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -48,67 +51,198 @@ class Hypre(Package):
+    variant('openmp', default=False, description='Use OpenMP')
+    variant('complex', default=False, description='Use complex values')
+    # TODO: Add 'cuda_arch' variant
+    # TODO: When are external superlu and superlu-dist needed or used?
+    # TODO: Add 'superlu-dist' variant? Can be used as coarse-grid solver in
+    #       v2.14.0 and later.
 
     depends_on("mpi", when='+mpi')
     depends_on("blas")
     depends_on("lapack")
+    depends_on('cuda@8:', when='+cuda')
+    # Is SuperLU/SuperLU_DIST required for +fei?
+    depends_on('superlu', when='@2.13.0: ~mpi +fei')
+    depends_on('superlu-dist', when='@2.13.0: +mpi +fei')
+
+    # GPU support was added in 2.13.0
+    conflicts('+cuda', when='@:2.12.99')
+    # The internal SuperLU was removed in 2.13.0
+    conflicts('+internal-superlu', when='@2.13.0:')
+    # TODO: handle '+cuda' with v2.14.0
+    conflicts('+cuda', when='@2.14.0', msg='todo: building 2.14.0 with cuda')
+
+    # Patch to add ppc64le in config.guess
+    patch('ibm-ppc64le.patch', when='@:2.11.1')
+    # Patch for building v2.13.0 with cuda
+    patch('hypre-2.13.0-cuda.patch', when='@2.13.0+cuda')
 
     def install(self, spec, prefix):
-        # Note: --with-(lapack|blas)_libs= needs space separated list of names
+        # Note: --with-(lapack|blas)-libs= needs space separated list of names
         lapack = spec['lapack'].libs
         blas = spec['blas'].libs
 
+        # TODO: C++ is only needed when 'fei' is enabled?
         configure_args = [
             '--prefix=%s' % prefix,
             '--with-lapack-libs=%s' % ' '.join(lapack.names),
             '--with-lapack-lib-dirs=%s' % ' '.join(lapack.directories),
             '--with-blas-libs=%s' % ' '.join(blas.names),
             '--with-blas-lib-dirs=%s' % ' '.join(blas.directories)
         ]
+        make_args = []
 
         if '+mpi' in self.spec:
-            os.environ['CC'] = spec['mpi'].mpicc
-            os.environ['CXX'] = spec['mpi'].mpicxx
-            os.environ['F77'] = spec['mpi'].mpif77
-            configure_args.append('--with-MPI')
+            configure_args += ['--with-MPI']
+            env['CC'] = spec['mpi'].mpicc
+            env['CXX'] = spec['mpi'].mpicxx
         else:
             configure_args.append('--without-MPI')
 
+        cflags = spec.compiler_flags['cflags']
+        cxxflags = spec.compiler_flags['cxxflags']
+        fflags = spec.compiler_flags['fflags']
+
+        if cflags:
+            configure_args += ['CFLAGS=%s' % ' '.join(cflags)]
+        if cxxflags:
+            configure_args += ['CXXFLAGS=%s' % ' '.join(cxxflags)]
+        if fflags:
+            configure_args += ['FCFLAGS=%s' % ' '.join(fflags)]
+
         if '+int64' in self.spec:
             configure_args.append('--enable-bigint')
 
         if '+shared' in self.spec:
             configure_args.append("--enable-shared")
 
-        if '~internal-superlu' in self.spec:
+        if '@:2.12.99~internal-superlu' in self.spec:
```

Comment:
since you have conflicts above, you can skip adding `version` constraint here as `+internal-superlu` will be possible for recent versions only.

Reply:
Note that this is `~internal-superlu` which has no conflicts.